### DISCUSSION TITLE
Does Piccolo ORM suffer of the problem of useless order by on some ORM prefetches ?

### DISCUSSION THREAD

**Discussion by LLyaudet** [piccolo-orm/piccolo#958] [Q&A] [answered]
Hello,

Recently I found a problem with Django and submitted a patch in a PR:
https://code.djangoproject.com/ticket/35309
https://github.com/django/django/pull/17984
I also developped a work-around for old versions of Django:
https://github.com/LLyaudet/django-monkey-patches/blob/main/src/django_monkey_patches/django__orm__prefetch_without_useless_order_by.py

I don't know yet Piccolo ORM details.
The idea is just that some models can have a default ordering (I assume this functionality is present in Piccolo ORM also).
And that some ORM may still add this ordering in the SQL query whilst it is not needed because only a single object will be prefetched per instance of the original result set.
On very big prefetches I noticed a gain of 10 - 15 %,
since the rest of the DB query is mainly an index scan.
#ClimateChangeBrake

Can someone please dump the queries of such prefetches and see if the bug is present in Piccolo ORM ?
If so submitting a PR should not be too hard.
I will try to help.
But right now, I start by warning the frameworks/ORM I am aware of :)

Best regards,
      Laurent Lyaudet

**Comment by sinisaos** [comment, marked as answer]
@LLyaudet Piccolo shouldn't have that problem because there is no default ordering and you have to explicitly use `order_by` method to sort the query results. Hope that helps.

**Reply by LLyaudet** [reply]
Thanks for your answer :)
